### PR TITLE
server: do not remove volume from DB if fail to expunge it from primary storage or secondary storage

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR fixes the issue that volume is removed from cloudstack DB but not removed from primary storage or secondary storage.

steps to reproduce the issue on kvm
(1) create a volume, attach to a vm, detach it
(2) stop all cloudstack-agent
(3) remove the volume.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

File: server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java [hunk 1]
```diff
@@ -1407,29 +1407,36 @@ protected void destroyVolumeIfPossible(VolumeVO volume) {
...
+            s_logger.info("Expunging volume " + volume.getId() + " from " + role + " data store");
+            AsyncCallFuture<VolumeApiResult> future = volService.expungeVolumeAsync(volOnStorage);
+            VolumeApiResult result = future.get();
+            if (result.isFailed()) {
+                s_logger.warn("Failed to expunge the volume " + volume + " in " + role + " data store");
+                String details = "";
+                if (result.getResult() != null && !result.getResult().isEmpty()) {
+                    details = result.getResult();
+                }
+                throw new CloudRuntimeException(details);
```
Comment (nvazquez) [inline server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java:1433]: Minor suggestion: what about prepending `"Failed to expunge the volume " + volume + " in " + role + " data store: "` here?

**Review by rhtyd — APPROVED** [review]

File: server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java (hunk 1 above)
Comment (rhtyd) [reply server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java:1433]: cc @weizhouapache

Comment (weizhouapache) [reply server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java:1434]: @nvazquez done, thanks.

**Review by nvazquez — APPROVED** [review]
LGTM

**Pull request merged**